wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java

```diff
@@ -1,51 +1,79 @@
 package com.dylibso.chicory.wasm.types;
 
-import static com.dylibso.chicory.wasm.types.ValueType.ID.ExnRef;
-
 import com.dylibso.chicory.wasm.MalformedException;
 import java.util.List;
 
 /**
  * The possible WASM value types.
  */
-public enum ValueType {
```

Comment:
This is the biggest breaking change in this PR, from the binary point of view it's going to break, still, the source compatibility is preserved.

I'm looking for advise from @evacchi and @chirino :
after we merge this one, should we bump the major or is it ok to bump only the minor version?

Reply:
even though it's a breaking ABI change I wouldn't call for a major version. Let's highlight it in the release notes.

Reply:
The more I think about this the more I think it's a bit risky.
`ValueType.XXX` is exposed and used by almost anyone using Chicory, and I'm afraid that this change would break code :cold_sweat:

Reply:
yeah.. expressions like `switch(valueType)` would break.

Reply:
this is a good point. Even refactoring within chicory seemed to break random places that did `valueType == ValueType.V128`. I think you had a question before about using .equals or comparing opcode @andreaTP , I think I'm now leaning towards always using opcodes to compare and removing the `ValueType.XXX` static constants

Reply:
mm yeah I didn't think about that :D

Reply:
> removing the ValueType.XXX static constants

I'm not sure this is a good idea, I really like being able to use static constants when possible.

That said, this means that we need to plan a ahead on how to make this work, the problem is that this is extensive and maintaining the diff in a branch is going to be a lot of work.
On the other side merging means that we need to release a 2.X as next.

I'll think a bit about this, happy to hear feedback!

Reply:
@hypercubestart sorry for the delay we brainstormed a bit on this during yesterday's meeting, I think I have a plan to try to make this change not break the API.

Reply:
Finalized the idea we had yesterday:
https://github.com/hypercubestart/chicory/pull/2

cc. @chirino @evacchi can you double-check that I have not missed something?

Reply:
resolved